### ISSUE TITLE
Mark tests requiring network access

### ISSUE THREAD

**Issue by mcepl** [googleapis/python-storage#457]
All builds (and running test suites as part of them) of OpenSUSE packages (the same for every other Linux or Mac distribution) happen in the chrooted environment without the network access, which makes running of the test suite for this package extremely awkward. Most of tests fail because they require network access, which is a bad idea even for other reasons (mocking is the way to go).

[no-network.patch](https://github.com/googleapis/python-storage/files/6618102/no-network.patch.txt) is the solution of my problem, marking all tests requiring network access appropriately, and then skipping so marked ones.
#### Environment details

  - OS type and version: any build system without network access
  - Python version: any (tested with Python 3.6-3.9)
  - pip version: any
  - `google-cloud-storage` version: 1.38.0 (from the tarball from PyPI)

**Comment by tseaver** [comment]
PR #416 tracks updating all unit tests such that they no longer require actual credentials to run:  none of them actually make use of the network, but the patches for the PRs tracked there ensure it (by mocking out the client).  Most are already merged, except for PR #459, which should merge soon.   If you'd like to test in your environment, try checking out the  `416-avoid_real_client_in_bucket_blob_unit_tests` [branch](https://github.com/googleapis/python-storage/tree/416-avoid_real_client_in_bucket_blob_unit_tests):  the unit tests run cleanly in my environment with no credentials configured.

*All* system tests, however, are intended to call the actual APIs -- they are how we ensure that the library does not drift away from the backend.

**Comment by mcepl** [comment]
> _All_ system tests, however, are intended to call the actual APIs -- they are how we ensure that the library does not drift away from the backend.

I understand that. However, I really have no network access in the build system, and I would appreciate even just a partial test suite to find out that the package is not completely broken.

**Comment by tseaver** [comment]
@mcepl You should be able to run the unit test suite without any network access:  I just did it with my laptop in "airplane mode" on the `416-avoid_real_client_in_bucket_blob_unit_tests` branch:

![Screenshot from 2021-06-09 11-01-42](https://user-images.githubusercontent.com/242750/121379967-843b3880-c912-11eb-89ee-db1b52fd37e3.png)

**Comment by mcepl** [comment]
Yes, that branch works offline as well, thank you. However, there is no way for me to include this into the official openSUSE unreleased, so I will wait on that. Thank you very much anyway.

**Comment by tseaver** [comment]
I expect that branch's PR to be merged shortly, with a new release to follow this week.

**Comment by tseaver** [comment]
@mcepl PR #459 has now merged.  We will be making a release no later than next Monday, 2021-06-14.
